File: server_node_backup/prisma/seed.ts
import { PrismaClient, Role } from '@prisma/client';
import * as bcrypt from 'bcryptjs';

const prisma = new PrismaClient();

async function main() {
    // 1. Create Bases
    console.log('Seeding Bases...');
    const baseAlpha = await prisma.base.upsert({
        where: { name: 'Fort Alpha' },
        update: {},
        create: {
            name: 'Fort Alpha',
            location: 'Nevada, USA',
        },
    });

    const baseBravo = await prisma.base.upsert({
        where: { name: 'Outpost Bravo' },
        update: {},
        create: {
            name: 'Outpost Bravo',
            location: 'Helmand, Afghanistan',
        },
    });

    // 2. Create Asset Types
    console.log('Seeding Asset Types...');
    const m4Carbine = await prisma.assetType.upsert({
        where: { name: 'M4 Carbine' },
        update: {},
        create: {
            name: 'M4 Carbine',
            description: 'Standard issue assault rifle',
        },
    });

    const m1Abrams = await prisma.assetType.upsert({
        where: { name: 'M1 Abrams' },
        update: {},
        create: {
            name: 'M1 Abrams',
            description: 'Main Battle Tank',
        },
    });

    const ammo556 = await prisma.assetType.upsert({
        where: { name: '5.56mm Ammo Crate' },
        update: {},
        create: {
            name: '5.56mm Ammo Crate',
            description: 'Crate containing 1000 rounds',
        },
    });

    // 3. Create Users
    console.log('Seeding Users...');
    const salt = await bcrypt.genSalt(10);
    const password = await bcrypt.hash('password123', salt);

    // Admin
    await prisma.user.upsert({
        where: { username: 'admin' },
        update: {},
        create: {
            username: 'admin',
            password,
            role: Role.ADMIN,
        },
    });

    // Commander Alpha
    await prisma.user.upsert({
        where: { username: 'commander_alpha' },
        update: {},
        create: {
            username: 'commander_alpha',
            password,
            role: Role.BASE_COMMANDER,
            baseId: baseAlpha.id,
        },
    });

    // Logistics Alpha
    await prisma.user.upsert({
        where: { username: 'logistics_alpha' },
        update: {},
        create: {
            username: 'logistics_alpha',
            password,
            role: Role.LOGISTICS_OFFICER,
            baseId: baseAlpha.id,
        },
    });

    // Commander Bravo
    await prisma.user.upsert({
        where: { username: 'commander_bravo' },
        update: {},
        create: {
            username: 'commander_bravo',
            password,
            role: Role.BASE_COMMANDER,
            baseId: baseBravo.id,
        },
    });

    console.log('Seeding complete.');
}

main()
    .catch((e) => {
        console.error(e);
        process.exit(1);
    })
    .finally(async () => {
        await prisma.$disconnect();
    });

File: server_fastapi_backup/seed.py
from app.database import SessionLocal, engine, Base
from app import models, dependencies
from sqlalchemy.exc import IntegrityError

db = SessionLocal()

def seed():
    print("Creating tables...")
    Base.metadata.create_all(bind=engine)

    print("Seeding Bases...")
    bases = [
        {"name": "Alpha Base", "location": "Sector 1"},
        {"name": "Bravo Base", "location": "Sector 2"}
    ]
    for b in bases:
        if not db.query(models.BaseObj).filter_by(name=b["name"]).first():
            db.add(models.BaseObj(**b))
    
    print("Seeding Asset Types...")
    asset_types = [
        {"name": "Rifle M4", "description": "Standard issue rifle"},
        {"name": "Night Vision Goggles", "description": "Gen 3 NVG"},
        {"name": "Tank M1", "description": "Main Battle Tank"}
    ]
    for a in asset_types:
        if not db.query(models.AssetType).filter_by(name=a["name"]).first():
            db.add(models.AssetType(**a))

    print("Seeding Users...")
    # Admin
    if not db.query(models.User).filter_by(username="admin").first():
        admin = models.User(
            username="admin",
            password=dependencies.get_password_hash("admin123"),
            role=models.Role.ADMIN
        )
        db.add(admin)
    
    # Commander
    base_alpha = db.query(models.BaseObj).filter_by(name="Alpha Base").first()
    if base_alpha and not db.query(models.User).filter_by(username="commander").first():
        commander = models.User(
            username="commander",
            password=dependencies.get_password_hash("commander123"),
            role=models.Role.BASE_COMMANDER,
            baseId=base_alpha.id
        )
        db.add(commander)

    try:
        db.commit()
        print("Seeding complete!")
    except IntegrityError as e:
        db.rollback()
        print(f"Error seeding data: {e}")
    finally:
        db.close()

if __name__ == "__main__":
    seed()
